commands/general/stats.js

```diff
@@ -4,118 +4,165 @@ const moment = require('moment')
 
 module.exports.load = (client) => {
   client.commands['stats'] = {
-    run (message) {
-      let args = message.content.split(' ').splice(1)
-      let username
-      let discriminator
-      let mem
-      let userId
-      let av = './assets/default_avatar.jpg'
+    async run (message) {
+      let userInfo
+      try {
+        userInfo = selectTargetUser(message)
+      } catch (err) {
+        console.log(err.stack)
+        return client.errorMessage(message, `Unable to identify user: ${err}`)
+      }
+
+      try {
+        const [user, pos, tot, guild] = await Promise.all([
+          client.userRepository.load(userInfo.userId),
+          client.userRepository.getSessionPos(userInfo.userId),
+          client.userRepository.getSessionCount(),
+          client.guildRepository.load(message.guild.id)
+        ])
+        if (user !== null) {
+          const activeTime = getActiveTime(guild, userInfo.mem)
 
-      if (args.length >= 1) {
-        // fqName: "fully qualified name"
-        let fqName = args.join(' ').trim().split('#')
-        if (fqName.length === 2) {
-          username = fqName[0]
-          discriminator = fqName[1]
-          mem = client.guilds.get(message.guild.id).members.find(val => val.user.username === username && val.user.discriminator === discriminator)
```

Comment:
I'm pretty sure spaces doesn't work (in this code or in the refactor) because I think discord.js keeps the `username` all lowercase with spaces removed.

Reply:
I'm not sure I understand this. Are you saying we can't get stats for users with a username that has a space in it right now? I am currently able to do that for the production bot.